### PR TITLE
[AMSDK-10267] Handle AAM response content events

### PR DESCRIPTION
**Pull request by rymorale**
## Description
The Audience Manager extension will dispatch a response content event indicating if the opt out hit sent by the extension was successful. If it is false the Identity extension will send the opt out hit.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPIdentity/Sources/Identity.swift [hunk 1]
```diff
@@ -147,6 +148,15 @@ import Foundation
         dispatch(event: responseEvent)
     }
 
+    /// Handles Audience Response Content events containing a flag which signals if the opt-out hit was sent by the Audience Extension.
+    /// If the flag is false, the Identity extension will send an opt-out hit to the configured Identity server.
+    /// - Parameter event: The event coming from the Audience Manager extension
+    private func handleAudienceResponse(event: Event) {
+        if let optOutHitSent = event.data?[IdentityConstants.Audience.OPTED_OUT_HIT_SENT] as? Bool, optOutHitSent == true { return }
```
Comment (nporter-adbe) [inline AEPIdentity/Sources/Identity.swift:155]: Let's add a log here stating we're ignoring the event when we return.

Comment (nporter-adbe) [inline AEPIdentity/Sources/Identity.swift:155]: Lets add this event data read in Event+Identity.

```swift
var optOutHitSent: Bool {
   return event.data?[IdentityConstants.Audience.OPTED_OUT_HIT_SENT] as? Bool ?? false
}
```

**Review by nporter-adbe — APPROVED** [review]
looks good 👍🏻

**Comment by rymorale** [comment]
Swift format formatted some files locally which then caused a format failure on CI. Am I using an incorrect version of swift format locally?

**Pull request merged**